theopensystemslab/planx-new
# feat: add audit table to track reconciled sessions

this audit table will let us do two key things:
- track how many times a save & return session is successfully resumed (group by session_id & count)
- better troubleshoot applicant issues for cases like [this one recently reported in #help-issues](https://opendigitalplanning.slack.com/archives/C0241GWFG4B/p1669724872840919) 

future scope: 
- delete `response` values when an associated lowcal_session is marked as deleted_at. this will let us keep session ids & resumed timestamps for analytics, but not retain personal session data. associated ticket here https://trello.com/c/FlGxMbps/2078-data-retention-create-single-scheduled-task-to-purge-application-data-at-set-time-interval
- create a proper foreign key relationship between `lowcal_sessions.id` & `reconciliation_requests.session_id` ? (reconciliation_requests follows pattern of other existing audit tables like planning_constraints_requests, bops_ & uniform_applications which means no upfront foreign keys and session ids are stored as text for simplicity)

to test: 
- go through a save & return flow
- save & resume your session
- confirm that an entry has been created in table `reconciliation_requests`
